src/TeaPie/Logging/RequestsLoggingHandler.cs

```diff
@@ -0,0 +1,165 @@
+﻿using System.Net.Http.Headers;
+using TeaPie.Http;
+using TeaPie.Http.Auth;
+using Microsoft.Extensions.Logging;
+
+namespace TeaPie.Logging;
+
+internal class RequestsLoggingHandler(IAuthProviderAccessor authProviderAccessor, ILoggerFactory loggerFactory) : DelegatingHandler
+{
+    private readonly IAuthProviderAccessor _authProviderAccessor = authProviderAccessor;
+    private readonly ILogger _logger = loggerFactory.CreateLogger("HttpRequests");
+    private static readonly HttpRequestOptionsKey<RequestExecutionContext> _contextKey = new("__TeaPie_Context__");
+    public static readonly HttpRequestOptionsKey<RequestLogFileEntry> LogEntryKey = new("__TeaPie_LogEntry__");
+
+    protected override async Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
+    {
+        if (!request.Options.TryGetValue(_contextKey, out var requestContext) || requestContext is null)
+        {
+            return await base.SendAsync(request, cancellationToken);
+        }
+
+        var logEntry = await GetOrCreateLogEntryAsync(request, requestContext);
+        var attemptStartTime = DateTime.UtcNow;
+        HttpResponseMessage? response = null;
+        Exception? exception = null;
+
+        try
+        {
+            response = await base.SendAsync(request, cancellationToken);
+            return response;
+        }
+        catch (Exception ex)
+        {
+            exception = ex;
```

Comment:
Toto sa mi moc nepáči.

Nie je dobrá prax si takto odložiť vec a inde ju použiť.

Aj samotný `try {} catch {} finaly {}` je robený na to aby si v ináč pristúpila k chybe a ináč k úspešnemu vykonaniu.

Sprav to radšej tak, že v `finally` sa nebude volať. Tieto metódy zavoláš v `catch` s tou exception a  normálne v `try` zavoláš tie metódy bez exception.

nebudeš si musieť takto odkladať premennú.